cmd/go-ouroboros-network/chainsync.go

```diff
@@ -0,0 +1,225 @@
+			h := blockData.(*block.ByronEpochBoundaryBlockHeader)
+			//fmt.Printf("era = Byron (EBB), epoch = %d, id = %s\n", h.ConsensusData.Epoch, h.Id())
+			if syncState.byronEpochSlot > 0 {
+				syncState.byronEpochBaseSlot += syncState.byronEpochSlot + 1
+			}
+			blockSlot = syncState.byronEpochBaseSlot
+			blockHash, _ = hex.DecodeString(h.Id())
+		case block.BLOCK_TYPE_BYRON_MAIN:
+			h := blockData.(*block.ByronMainBlockHeader)
+			//fmt.Printf("era = Byron, epoch = %d, slot = %d, id = %s\n", h.ConsensusData.SlotId.Epoch, h.ConsensusData.SlotId.Slot, h.Id())
+			syncState.byronEpochSlot = uint64(h.ConsensusData.SlotId.Slot)
+			blockSlot = syncState.byronEpochBaseSlot + syncState.byronEpochSlot
+			blockHash, _ = hex.DecodeString(h.Id())
+		default:
+			h := blockData.(*block.ShelleyBlockHeader)
+			blockSlot = h.Body.Slot
+			blockHash, _ = hex.DecodeString(h.Id())
+		}
+		if err := syncState.oConn.BlockFetch.RequestRange([]interface{}{blockSlot, blockHash}, []interface{}{blockSlot, blockHash}); err != nil {
+			fmt.Printf("error calling RequestRange: %s\n", err)
+			return err
+		}
+	} else {
+		switch blockType {
+		case block.BLOCK_TYPE_BYRON_EBB:
+			b := blockData.(*block.ByronEpochBoundaryBlock)
+			fmt.Printf("era = Byron (EBB), epoch = %d, id = %s\n", b.Header.ConsensusData.Epoch, b.Id())
+		case block.BLOCK_TYPE_BYRON_MAIN:
+			b := blockData.(*block.ByronMainBlock)
+			fmt.Printf("era = Byron, epoch = %d, slot = %d, id = %s\n", b.Header.ConsensusData.SlotId.Epoch, b.Header.ConsensusData.SlotId.Slot, b.Id())
+		case block.BLOCK_TYPE_SHELLEY:
+			b := blockData.(*block.ShelleyBlock)
+			fmt.Printf("era = Shelley, slot = %d, block_no = %d, id = %s\n", b.Header.Body.Slot, b.Header.Body.BlockNumber, b.Id())
+		case block.BLOCK_TYPE_ALLEGRA:
+			b := blockData.(*block.AllegraBlock)
+			fmt.Printf("era = Allegra, slot = %d, block_no = %d, id = %s\n", b.Header.Body.Slot, b.Header.Body.BlockNumber, b.Id())
+		case block.BLOCK_TYPE_MARY:
+			b := blockData.(*block.MaryBlock)
+			fmt.Printf("era = Mary, slot = %d, block_no = %d, id = %s\n", b.Header.Body.Slot, b.Header.Body.BlockNumber, b.Id())
+		case block.BLOCK_TYPE_ALONZO:
+			b := blockData.(*block.AlonzoBlock)
+			fmt.Printf("era = Alonzo, slot = %d, block_no = %d, id = %s\n", b.Header.Body.Slot, b.Header.Body.BlockNumber, b.Id())
+		default:
+			fmt.Printf("unsupported (so far) block type %d\n", blockType)
+			fmt.Printf("%s\n", utils.DumpCborStructure(blockData, ""))
+		}
+		syncState.readyForNextBlockChan <- true
+	}
+	return nil
+}
+
+func chainSyncIntersectFoundHandler(point interface{}, tip interface{}) error {
+	fmt.Printf("found intersect: point = %#v, tip = %#v\n", point, tip)
+	syncState.readyForNextBlockChan <- true
+	return nil
+}
+
+func chainSyncIntersectNotFoundHandler() error {
+	fmt.Printf("ERROR: failed to find intersection\n")
+	os.Exit(1)
+	return nil
+}
+
+func chainSyncDoneHandler() error {
+	return nil
+}
+
+func blockFetchStartBatchHandler() error {
+	return nil
```

Comment:
I've seen a few of these now.  Are they just placeholders for future development?

Reply:
The library will (purposely) blow up if you don't define a callback function and receive the associated message, but we don't really care about the message here, so the callback does nothing.